.github/workflows/codeql-analysis.yml

```diff
@@ -0,0 +1,44 @@
+---
+name: CodeQL
+
+# Cancel previous workflow run groups that have not completed.
+concurrency:
+  # Group workflow runs by workflow name, along with the head branch ref of the pull request
+  # or otherwise the branch or tag ref.
+  group: ${{ github.workflow }}-${{ github.event_name == 'pull_request' && github.head_ref || github.ref }}
+  cancel-in-progress: true
+
+# Disable permissions for all available scopes.
+# Enable permissions for specific scopes as needed on job level.
+permissions: {}
+
+on:
+  push:
+    # Only run if JS files changed.
+    paths:
+      - '**.js'
+    branches:
+      - trunk
+      - 'release/**'
+  pull_request:
+    # Only run if JS files changed.
+    paths:
+      - '**.js'
+
+jobs:
+  analyze:
+    name: Analyze
+    runs-on: ubuntu-latest
+    permissions:
+      security-events: write
```

Comment:
Why is this needed?

Reply:
Since CodeQL requires permission to write to security events as per https://github.com/github/codeql-action?tab=readme-ov-file#permissions

More info on how it's add to security events - https://github.com/WordPress/performance/security/code-scanning?query=pr%3A981+is%3Aopen